### PR TITLE
React2 week1 homework

### PR DESCRIPTION
**Pull request by YF91925352**

### CODE REVIEW

File: src/client/App.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
-        <p>lol</p>
+        <Home />
       </Route>
       <Route exact path="/test-component">
         <TestComponent></TestComponent>
       </Route>
+      <Route exact path="/mealist">
+        <Mealist />
+      </Route>
+      <Route exact path="/mealcard">
```
Comment (shpomp) [inline src/client/App.js:20]: Good to practise, but you will not use this, as you will need a rout with a path parameter 👍🏽

File: src/client/components/MealProvider.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React, { useState, createContext, useEffect } from "react";
+export const mealContext = createContext();
+export function MealProvider({ children }) {
+  const [list, setList] = useState([]);
```
Comment (shpomp) [inline src/client/components/MealProvider.jsx:4]: A more descriptive name would be better 👌🏽

File: src/client/components/Mealist.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React, { useState, useEffect, useContext } from "react";
+import { mealContext } from "./MealProvider";
+export default function Mealist() {
```
Comment (shpomp) [inline src/client/components/Mealist.jsx:3]: camelCase for naming ⚠️

File: src/client/components/Mealcard.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { mealContext } from "./MealProvider";
+import "../mealCard.css";
+export default function Mealcard() {
+  const { list } = useContext(mealContext);
+
+  return (
+    <div className="meal-list">
+      <h1>Meal Card</h1>
+      <div>
+        {list.map((item) => (
```
Comment (shpomp) [inline src/client/components/Mealcard.jsx:11]: If this is the single meal card, why is it mapping over all the meals, which is already done in MealList?

**Review by shpomp — CHANGES_REQUESTED** [review]
Nice! 👍🏽 
Some structural and logical changes, but some of the basics are very good! ⭐

File: src/client/components/Mealcard.jsx (hunk 4 above)
Comment (YF91925352) [reply src/client/components/Mealcard.jsx:11]: Thx for your review!I will update them!

**Pull request merged**